### PR TITLE
Feature: Renamed "Favorites" section to "Pinned"

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

- Replaced all "Favorite" & "favorite" entities.
- No "Favourites" & "favourite" entity found.

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
- [ ] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [x] Did you remove any strings from the en-us resource file?
   - [x] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Launch the app
   2. See the sidebar and "Pinned" string is shown

## Screenshots

_None_

### CODE REVIEW

File: docs/rich-commands.md
Comment (yaira2) [inline docs/rich-commands.md:1]: Can you also update the actions page on the website?

File: src/Files.App/ResourceDictionaries/PathIcons.xaml [hunk 1]
```diff
@@ -1448,7 +1448,7 @@
 				</Setter>
 			</Style>
 
-			<Style x:Key="ColorIconPinToFavorites" TargetType="local:OpacityIcon">
+			<Style x:Key="ColorIconPinToSidebar" TargetType="local:OpacityIcon">
```
Comment (yaira2) [inline src/Files.App/ResourceDictionaries/PathIcons.xaml:1451]:
```suggestion
			<Style x:Key="ColorIcon.PinToSidebar.16x16" TargetType="local:OpacityIcon">
```

File: src/Files.App/ResourceDictionaries/PathIcons.xaml [hunk 2]
```diff
@@ -1489,7 +1489,7 @@
 				</Setter>
 			</Style>
 
-			<Style x:Key="ColorIconUnpinFromFavorites" TargetType="local:OpacityIcon">
+			<Style x:Key="ColorIconUnpinFromSidebar" TargetType="local:OpacityIcon">
```
Comment (yaira2) [inline src/Files.App/ResourceDictionaries/PathIcons.xaml:1492]:
```suggestion
			<Style x:Key="ColorIcon.UnpinFromSidebar.16x16" TargetType="local:OpacityIcon">
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 3]
```diff
@@ -1158,8 +1158,8 @@
   <data name="DrivesWidgetOptionsFlyoutMapNetDriveMenuItem.Text" xml:space="preserve">
     <value>Map network drive</value>
   </data>
-  <data name="SidebarFavorites" xml:space="preserve">
-    <value>Favorites</value>
+  <data name="SidebarPinned" xml:space="preserve">
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:1161]:
```suggestion
  <data name="Pinned" xml:space="preserve">
```
Make sure to also rename the references to this string

File: docs/rich-commands.md
Comment (0x5bfa) [reply docs/rich-commands.md:1]: Sure.
Can I remove this markdown? This duplicates work for those who work on Actions.
Or as another idea you could enable Wiki as a pool for the website update

Comment (yaira2) [reply docs/rich-commands.md:1]: We can remove the markdown, it was mainly intended to track our progress on rich commands but it's no longer needed.

Comment (0x5bfa) [reply docs/rich-commands.md:1]: I'll remove in an upcoming PR.

File: src/Files.App/ResourceDictionaries/PathIcons.xaml (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/ResourceDictionaries/PathIcons.xaml]: Done.

File: src/Files.App/ResourceDictionaries/PathIcons.xaml (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App/ResourceDictionaries/PathIcons.xaml]: Done

File: src/Files.App/ResourceDictionaries/PathIcons.xaml [hunk 4]
```diff
@@ -1448,7 +1448,7 @@
 				</Setter>
 			</Style>
 
-			<Style x:Key="ColorIconPinToSidebar" TargetType="local:OpacityIcon">
+			<Style x:Key="ColorIcon.PinToSidebar.16x16" TargetType="local:OpacityIcon">
```
Comment (yaira2) [inline src/Files.App/ResourceDictionaries/PathIcons.xaml:1451]: I didn't realize this was also used for "Pin to start". Maybe it should just be "Pin" and "Unpin".

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**